### PR TITLE
Some clean-up of buildslave unit tests,

### PR DESCRIPTION
**Pull request by elmirjagudin**
* use @defer.inlineCallbacks style
* added proper skip decarators for win32 only tests
* drop python 2.4 hacks

### CODE REVIEW

File: slave/buildslave/test/unit/test_commands_bk.py [hunk 1]
```diff
@@ -63,7 +65,7 @@ def test_simple(self):
         ]
         self.patch_runprocess(*expects)
 
-        d = self.run_command()
+        yield self.run_command()
+
         # TODO: why the extra quotes?
-        d.addCallback(self.check_sourcedata, '"http://bkdemo.bkbits.net/bk_demo1\n"')
-        return d
+        self.assertSourceData('"http://bkdemo.bkbits.net/bk_demo1\n"')
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_bk.py]: While you are at it, could you please fix the domain name to something `example`?  E.g. `bkdemo.example.net`?

File: slave/buildslave/test/unit/test_commands_bzr.py [hunk 2]
```diff
@@ -72,6 +74,7 @@ def test_simple(self):
         ]
         self.patch_runprocess(*expects)
 
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "http://bazaar.launchpad.net/~bzr/bzr-gtk/trunk\n")
-        return d
+        yield self.run_command()
+
+        self.assertSourceData(
+            "http://bazaar.launchpad.net/~bzr/bzr-gtk/trunk\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_bzr.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_cvs.py [hunk 3]
```diff
@@ -57,7 +59,7 @@ def test_simple(self):
...
         self.patch_runprocess(*expects)
 
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata,
-                      ":pserver:anoncvs@anoncvs.NetBSD.org:/cvsroot\nhtdocs\nNone\n")
-        return d
+        yield self.run_command()
+
+        self.assertSourceData(
+            ":pserver:anoncvs@anoncvs.NetBSD.org:/cvsroot\nhtdocs\nNone\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_cvs.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_darcs.py [hunk 4]
```diff
@@ -63,9 +65,9 @@ def test_simple(self):
         ]
         self.patch_runprocess(*expects)
 
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "http://darcs.net\n")
-        return d
+        yield self.run_command()
+
+        self.assertSourceData("http://darcs.net\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_darcs.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_git.py [hunk 5]
```diff
@@ -88,10 +89,12 @@ def test_run_mode_copy_fresh_sourcedir(self):
         ]
         self.patch_runprocess(*expects)
 
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "git://github.com/djmitche/buildbot.git master\n")
-        return d
+        yield self.run_command()
 
+        self.assertSourceData(
+            "git://github.com/djmitche/buildbot.git master\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_git.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_git.py [hunk 6]
```diff
@@ -138,10 +141,12 @@ def test_run_mode_copy_update_sourcedir(self):
         ]
         self.patch_runprocess(*expects)
 
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "git://github.com/djmitche/buildbot.git master\n")
-        return d
+        yield self.run_command()
+
+        self.assertSourceData(
+            "git://github.com/djmitche/buildbot.git master\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_git.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_git.py [hunk 7]
```diff
@@ -267,10 +272,12 @@ def test_run_mode_update_fresh(self):
...
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_git.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_git.py [hunk 8]
```diff
@@ -337,10 +345,12 @@ def test_run_mode_update_existing_known_rev(self):
         ]
         self.patch_runprocess(*expects)
 
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "git://github.com/djmitche/buildbot.git master\n")
-        return d
+        yield self.run_command()
 
+        self.assertSourceData(
+            "git://github.com/djmitche/buildbot.git master\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_git.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_git.py [hunk 9]
```diff
@@ -307,10 +314,11 @@ def test_run_mode_update_existing(self):
         ]
         self.patch_runprocess(*expects)
 
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "git://github.com/djmitche/buildbot.git master\n")
-        return d
+        yield self.run_command()
+        self.assertSourceData(
+            "git://github.com/djmitche/buildbot.git master\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_git.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_git.py [hunk 10]
```diff
@@ -381,10 +391,12 @@ def test_run_mode_update_existing_unknown_rev(self):
         ]
         self.patch_runprocess(*expects)
 
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "git://github.com/djmitche/buildbot.git master\n")
-        return d
+        yield self.run_command()
+
+        self.assertSourceData(
+            "git://github.com/djmitche/buildbot.git master\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_git.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_git.py [hunk 11]
```diff
@@ -435,10 +447,12 @@ def test_run_with_reference(self):
         ]
         self.patch_runprocess(*expects)
 
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "git://github.com/djmitche/buildbot.git master\n")
-        return d
+        yield self.run_command()
 
+        self.assertSourceData(
+            "git://github.com/djmitche/buildbot.git master\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_git.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_git.py [hunk 12]
```diff
@@ -473,10 +487,12 @@ def test_run_with_shallow_and_rev(self):
         ]
         self.patch_runprocess(*expects)
 
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "git://github.com/djmitche/buildbot.git master\n")
-        return d
+        yield self.run_command()
+
+        self.assertSourceData(
+            "git://github.com/djmitche/buildbot.git master\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_git.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_git.py [hunk 13]
```diff
@@ -523,10 +539,12 @@ def test_run_with_shallow(self):
...
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_git.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_git.py [hunk 14]
```diff
@@ -581,10 +599,12 @@ def test_run_with_shallow_and_reference(self):
         ]
         self.patch_runprocess(*expects)
 
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "git://github.com/djmitche/buildbot.git master\n")
-        return d
+        yield self.run_command()
 
+        self.assertSourceData(
+            "git://github.com/djmitche/buildbot.git master\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_git.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_git.py [hunk 15]
```diff
@@ -642,9 +662,10 @@ def test_run_with_submodules(self):
         ]
         self.patch_runprocess(*expects)
 
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "git://github.com/djmitche/buildbot.git master\n")
-        return d
+        yield self.run_command()
+
+        self.assertSourceData(
+            "git://github.com/djmitche/buildbot.git master\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_git.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_hg.py [hunk 16]
```diff
@@ -83,10 +85,11 @@ def test_simple(self):
         ]
         self.patch_runprocess(*expects)
 
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "http://bitbucket.org/nicolas17/pyboinc\n")
-        return d
+        yield self.run_command()
 
+        self.assertSourceData("http://bitbucket.org/nicolas17/pyboinc\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_hg.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_hg.py [hunk 17]
```diff
@@ -131,10 +134,12 @@ def test_update_existing(self):
             + 0,
         ]
         self.patch_runprocess(*expects)
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "http://bitbucket.org/nicolas17/pyboinc\n")
-        return d
 
+        yield self.run_command()
+
+        self.assertSourceData("http://bitbucket.org/nicolas17/pyboinc\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_hg.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_hg.py [hunk 18]
```diff
@@ -186,10 +191,12 @@ def test_update_existing_change_branch(self):
 
         ]
         self.patch_runprocess(*expects)
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "http://bitbucket.org/nicolas17/pyboinc\n")
-        return d
 
+        yield self.run_command()
+
+        self.assertSourceData("http://bitbucket.org/nicolas17/pyboinc\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_hg.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_hg.py [hunk 19]
```diff
@@ -242,10 +249,12 @@ def test_update_handle_emptyupdate(self):
             + 0,
         ]
         self.patch_runprocess(*expects)
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "http://bitbucket.org/nicolas17/pyboinc\n")
-        return d
 
+        yield self.run_command()
+
+        self.assertSourceData("http://bitbucket.org/nicolas17/pyboinc\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_hg.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_hg.py [hunk 20]
```diff
@@ -322,6 +331,7 @@ def test_update_existing_change_branch_purge_fail(self):
 
         ]
         self.patch_runprocess(*expects)
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "http://bitbucket.org/nicolas17/pyboinc\n")
-        return d
+
+        yield self.run_command()
+
+        self.assertSourceData("http://bitbucket.org/nicolas17/pyboinc\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_hg.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_p4.py [hunk 21]
```diff
@@ -98,12 +100,13 @@ def test_simple(self):
...
         self.patch_runprocess(*expects)
 
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "['p4dserv:1666', 'buildbot_test_10', " +
-                      "'//mydepot/myproj/', 'mytrunk', [], None, %s, 'copy', 'workdir']"
-                      % repr(self.basedir))
-        return d
+        yield self.run_command()
+        self.assertSourceData(
+            "['p4dserv:1666', 'buildbot_test_10', " +
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_p4.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_p4.py [hunk 22]
```diff
@@ -175,12 +178,11 @@ def test_simple_unicode_args(self):
...
-                      "['p4dserv:1666\\xe2\\x98\\x83', "
-                      "'buildbot_test_10\\xe2\\x98\\x83', "
-                      "'//mydepot/myproj/\\xe2\\x98\\x83', "
-                      "'mytrunk\\xe2\\x98\\x83', [], None, %s, 'copy', "
-                      "'workdir']"
-                      % repr(self.basedir))
-        return d
+        yield self.run_command()
+        self.assertSourceData(
+            "['p4dserv:1666\\xe2\\x98\\x83', "
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_p4.py]: Ditto.

File: slave/buildslave/test/unit/test_commands_svn.py [hunk 23]
```diff
@@ -64,9 +66,9 @@ def test_simple(self):
         ]
         self.patch_runprocess(*expects)
 
-        d = self.run_command()
-        d.addCallback(self.check_sourcedata, "http://svn.local/app/trunk\n")
-        return d
+        yield self.run_command()
+
+        self.assertSourceData("http://svn.local/app/trunk\n")
```
Comment (sa2ajj) [inline slave/buildslave/test/unit/test_commands_svn.py]: Ditto.

**Comment by sa2ajj** [comment]
Otherwise, very good, thank you!

(I was bitten a couple of times by using "real" domain names.)

**Comment by elmirjagudin** [comment]
Yeah, good idea to use example domains.

I'll try to get it done on the coming days.

**Comment by tardyp** [comment]
This is good work, but...

I think the plan is to drop support for slave based source steps. So I think I would prefer a PR that drops this code.
test-cmd-cleanups -> test-cmd-wipeout. :)

**Comment by tardyp** [comment]
Hi Elmir,
Do you intend to finish this in the next few days?

**Comment by elmirjagudin** [comment]
Oh sorry, forgot about this PR.

Can we drop slave based source steps right now? I can make PR which does that. 

Should all source commands be removed in slave/buildslave/commands?

Is this correct list of commands to be removed: cvs.py git.py bk.py darcs.py hg.py p4.py bzr.py svn.py ?

**Comment by tardyp** [comment]
Yes with removal of master side old source steps and all the associated
tests. Probably we first need a heads up on the mailing list.

**Comment by sa2ajj** [comment]
The slave-side source checkout steps are dropped already in master.

**Comment by elmirjagudin** [comment]
Ok, started work on removing slave-side source checkout steps: https://github.com/elmirjagudin/buildbot/tree/rm-slave-src-cmds

I'll make a PR once I have done more testing. We can send heads up in the mailing list when we are ready to merge it.

Closing this PR for now, I'll salvage the usefull bits into new PR once we are done with removing source commands.

**Pull request closed without merging**